microsoftgraph/microsoft-graph-explorer-v4
# Feature: Add NPS feedback and enable feedback collection compliance

## Overview
This PR: 
 - Adds the feedback button
 - Adds the NPS feedback, that is triggered automatically based on user activity
 - Ensures we collect the Age Group value for MSA users (GDPR Compliance)
 - Ensures we collect the tenant policies specified for an enterprise user (GDPR compliance)


### Demo
Feedback button
![image](https://user-images.githubusercontent.com/18407044/138898659-bfd3e849-5ea2-4ae1-8cff-5e34ba1ab28b.png)

Feedback pop-up
![image](https://user-images.githubusercontent.com/18407044/138899828-bfa00033-ef15-41e9-99a0-7cab364a8e88.png)

## Testing Instructions
_Testing the feedback button_
* Navigate to the "Got Feedback" tab and select it.
* Fill in the feedback form
* Attempt the same using different accounts (i.e AAD and MSA)

_Testing the NPS pop up_
* Load Graph Explorer and interact with the page for a minute (this is important, you must be active on the page for longer than a minute)
* Navigate out of the page (e.g open a new tab) 
* Return to Graph Explorer and interact with a component
* Observe the pop up appears requesting your feedback
* Fill in the form and submit
